crates/next-core/src/router.rs

```diff
@@ -342,12 +343,12 @@ pub async fn route(
         AssetIdentVc::from_path(project_path),
         context,
         intermediate_output_path,
-        Some(next_config),
+        None,
         vec![
             JsonValueVc::cell(request),
             JsonValueVc::cell(dir.to_string_lossy().into()),
         ],
-        routes_changed,
+        CompletionsVc::all(vec![next_config_changed, routes_changed]),
```

Comment:
Unrelated to the current PR, but it feels like `::all` should be a static method on `CompletionVc` instead of `CompletionsVc`.